### PR TITLE
Changed conditionally addition of type to open in case of _monitoring

### PR DESCRIPTION
**Pull request by andsel**
Changed conditionally addition of type to open in case of _monitoring endpoint
Related to LS [#11312](https://github.com/elastic/logstash/issues/11312) and plugin issue #899

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
+## 10.2.3
+  - Fixed 8.x type removal compatibility issue to continue for monitoring endpoint from LogStash's X-Pack[#899](https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/899)
```
Comment (robbavey) [inline CHANGELOG.md]: I'm not sure this review comment accurately describes the change? I think the description given in #899 is more apt.

Comment (andsel) [reply CHANGELOG.md]: Done, reworded a little bit. I hope it's not too long

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
+## 10.2.3
+  - After remove of type with [#892](https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/892) LogStash X-Pack monitoring started to fail with NPE, (see LogStash #11312) the cause is the ES endpoint used to push monitoring data (\_monitoring/) that needed the type field. Extracted the logic to decide if the type must be removed or not in a separate method and extend the plugin in X-Pack redefining the same method.
```
Comment (robbavey) [inline CHANGELOG.md]: I wonder if we could do something like:

```
Opened type removal logic for extension. This allows X-Pack Elasticsearch output to continue using types for special case `/_monitoring` bulk endpoint, enabling a fix for LogStash #11312. [#900](https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/900)
```

Comment (andsel) [reply CHANGELOG.md]: I like it, done

**Pull request closed without merging**